File: Solutions/102. Binary Tree Level Order Traversal/102. Binary Tree Level Order Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> queue=new LinkedList<>();
        var res=new LinkedList<List<Integer>>();
        if(root==null)
            return res;
        queue.offer(root);
        while(!queue.isEmpty()){
            int levelNum=queue.size();
            var tmp=new ArrayList<Integer>();
            for(int i=0;i<levelNum;++i){
                if(queue.peek().left!=null)
                    queue.offer(queue.peek().left);
                if(queue.peek().right!=null)
                    queue.offer(queue.peek().right);
                tmp.add(queue.poll().val);
            }
         res.add(tmp);   
        }
        return res;
    }
}

//DFS
class Solution {
    private List<List<Integer>> res=new ArrayList<>();
    public List<List<Integer>> levelOrder(TreeNode root){
        dfs(root,0);
        return res;
    }
    private void dfs(TreeNode root,int depth){
        if(root==null)
        return;
        if(res.size()==depth)
            res.add(new ArrayList<>());
        res.get(depth).add(root.val);
        dfs(root.left,depth+1);
        dfs(root.right,depth+1);
    }
}

File: Solutions/102. Binary Tree Level Order Traversal/102. Binary Tree Level Order Traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 //BFS
func levelOrder(root *TreeNode) [][]int {
    var q []*TreeNode
    var res [][]int
    if root==nil{
        return res
    }
    q=append(q,root)
    for len(q)>0{
        levelNum:=len(q)
        var tmp []int
        for i:=0;i<levelNum;i+=1{
            if q[0].Left!=nil{
                q=append(q,q[0].Left)
            }
            if q[0].Right!=nil{
                q=append(q,q[0].Right)
            }
            tmp=append(tmp,q[0].Val)
            q=q[1:]
        }
        res=append(res,tmp)
    }
    return res
}